internal/initializer/crds.go

```diff
@@ -131,6 +135,35 @@ func (c *CoreCRDs) Run(ctx context.Context, kube client.Client) error {
 		if err := pa.Apply(ctx, crd); err != nil {
 			return errors.Wrap(err, "cannot apply crd")
 		}
+
+		pendingCRDs[crd.Name] = struct{}{}
+	}
+
+	// wait for crds to be established
+	pollInterval := 2 * time.Second
+	timeout := 2 * time.Minute
+	if err := wait.PollUntilContextTimeout(ctx, pollInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		for crdName := range pendingCRDs {
+			crd := &extv1.CustomResourceDefinition{}
+			err := kube.Get(ctx, client.ObjectKey{Name: crdName}, crd)
+			if err != nil {
+				if kerrors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, errors.Wrapf(err, "failed to get CRD %s", crdName)
+			}
+
+			for _, cond := range crd.Status.Conditions {
+				if cond.Type == extv1.Established {
+					delete(pendingCRDs, crdName)
+					break
+				}
+			}
+		}
+		return len(pendingCRDs) == 0, nil
+	}); err != nil {
+		return errors.Wrap(err, "error waiting for CRDs to be established")
```

Comment:
Nit: include the pending CRDs in the error?

Reply:
Done